[docker-engine] Fix "no list of supported releases" (#5814)
captn3m0/endoflife.date: products/docker-engine.md

  -   git: https://github.com/moby/moby.git
      regex: ^v(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)(-ce)?$

> # For EOL, see https://github.com/moby/moby/blob/master/project/BRANCHES-AND-TAGS.md

Reviewer: The `releasePolicyLink`. must be updated with https://github.com/moby/moby/blob/master/project/BRANCHES-AND-TAGS.md too.

-   releaseCycle: "25.0"
    releaseDate: 2024-01-19

>     eol: 2024-07-25

Reviewer 2: The 25.0 branch is not going to be EOL for a couple of years, with Mirantis and Amazon employees contributing backports and CVE fixes

Author: Could you update https://github.com/moby/moby/blob/master/project/BRANCHES-AND-TAGS.md

Reviewer 2: https://github.com/moby/moby/pull/48474

-   releaseCycle: "23.0"
    releaseDate: 2023-02-02

>     eol: 2025-02-01 # estimated, see https://github.com/moby/moby/blob/master/project/BRANCHES-AND-TAGS.md

Reviewer 2: Mirantis will be maintaining the 23.0 branch until 2025-05-19.  https://docs.mirantis.com/mcr/23.0/compat-matrix/maintenance-lifecycle.html#mirantis-container-runtime-mcr

Author: Could you update https://github.com/moby/moby/blob/master/project/BRANCHES-AND-TAGS.md

> configuration files.

Docker Engine is supported by the [Moby Community](https://docs.docker.com/engine/install/#support),

> and as such.

Reviewer: ```suggestion
```

> The list of the supported releases can be found at <https://github.com/moby/moby/blob/master/project/BRANCHES-AND-TAGS.md>.

Reviewer: ```suggestion
The list of all supported releases is documented at <https://github.com/moby/moby/blob/master/project/BRANCHES-AND-TAGS.md>.
```

Minor rephrasing.
